src/features/cluster/delete-dialog/main/clear-as-deleting-handler.injectable.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { RequestChannelHandler } from "../../../../common/utils/channel/request-channel-listener-injection-token";
+import clustersThatAreBeingDeletedInjectable from "../../../../main/cluster/are-being-deleted.injectable";
+import type { ClearClusterAsDeletingChannel } from "../common/clear-as-deleting-channel";
+
+const clearClusterAsDeletingHandlerInjectable = getInjectable({
+  id: "clear-cluster-as-deleting-handler",
+  instantiate: (di): RequestChannelHandler<ClearClusterAsDeletingChannel> => {
```

Comment:
Maybe "the bite" is this: Here typing of instantiate is used to define the "contract" of injectable. If listener and handler of a channel were the same thing, the contract would be the familiar `injection token`, and this would look uniform.

Reply:
The contract of injection tokens are fine (and even used) but the problem with relying on them is that they are not type safe if the handler is supposed to have some inferred types from the channel

Reply:
> they are not type safe

For understanding: types in handler were inferred/derived from channel before, how were they not type safe?

Reply:
> types in handler were inferred/derived from channel before, how were they not type safe?

Incorrect, the shape of the injectable was inferred but not the param types or even the return type. Those had to be done by hand.

Reply:
That sounds like something we would not do. Weird. Maybe something happened in between.